src/coreclr/pal/src/init/pal.cpp

```diff
@@ -120,7 +120,7 @@ BOOL g_useDefaultBaseAddr = FALSE;
 
 /* critical section to protect access to init_count. This is allocated on the
    very first PAL_Initialize call, and is freed afterward. */
-static PCRITICAL_SECTION init_critsec = NULL;
+static PCRITICAL_SECTION init_critsec = nullptr;
```

Comment:
Reverting this line should fix the CG2 error.

Reply:
pushed